spectral_cubes/emission_search_cubes.md

```diff
@@ -0,0 +1,1101 @@
+```
+
+...and run `check_points_in_polygon` on the rows of our observations table in parallel. To do that, we'll use `multiprocessing`'s `map` method, which allows us to apply a function in parallel to each item (in this case, a row of `jwst_obstable` representing an observation) in an iterable. To improve performance, we'll first convert `jwst_obstable` to a `pandas` dataframe, and use the latter's `iterrows` method to create the iterable.
+
+```{code-cell} ipython3
+jwst_obstable_pandas = jwst_obstable.to_pandas()
+```
+
+If you examine the `Kernel usage` tab to your right while running this next cell, you'll see that you're now using nearly 100% of your multi-CPU capacity!
+
+```{code-cell} ipython3
+# Apply check_points_in_polygon to each item in the iterable, in parallel
+
+# In this case, since yso_coords will be the same for each row of the observations table,
+# we get the best performance if we define a one-time-use function as follows:
+def temp_apply_check_points(obs):
+    return check_points_in_polygon(obs, yso_coords)
+
+# Apply in parallel:
+with Pool(num_workers) as pool:
+    results = pool.map(temp_apply_check_points, [obs for _, obs in jwst_obstable_pandas.iterrows()])
+```
+
+```{code-cell} ipython3
+# Convert back to a pandas dataframe
+results = pd.DataFrame(results)
+
+# Remove empty rows, where check_points_in_polygon reached the "else" part of the return statement.
+results = results.dropna(how='all')
+
+# Convert back to an astropy table.
+yso_jwst_obstable = Table.from_pandas(results)
+
+print(f'We found {len(yso_jwst_obstable)} JWST spectral cube observations whose footprints overlap with YSO coordinates.')
+```
+
+## 2. Finding Fe II emission jets
+
++++
+
+We're looking for a way to identify jets exhibiting Fe II emission, without human intervention and without knowing much about the data quality or target source in each image. Let's first devise our approach by looking at a case example, and then we'll try it out on the full set of YSO JWST spectral cubes.
+
++++
+
+### 2.1 Retrieving sample data from AWS
+
++++
+
+From [Assani et al. 2025](https://arxiv.org/pdf/2504.02136), we know that the YSO called TMC1A features an extended jet with strong Fe II emission. Let's grab its coordinates from SIMBAD...
+
+```{code-cell} ipython3
+tmc1a_meta = Simbad.query_object('TMC1A')
+tmc1a_ra = float(tmc1a_meta['ra'])  # ICRS RA of TMC1A in degrees
+tmc1a_dec = float(tmc1a_meta['dec'])  # ICRS Dec of TMC1A in degrees
+```
+
+...and retrieve its observations from our table of JWST spectral cube observations of YSOs:
+
+```{code-cell} ipython3
+tmc1a_jwst = []
+
+# As before...
+for obs in yso_jwst_obstable:
+    polygon = parse_polygon(obs['s_region'])
+    polygon_path = Path(polygon, closed=True)
+    # We'll use matplotlib.Path.contains_point (singular) this time,
+    # instead of .contains_points (plural)
+    is_inside = polygon_path.contains_point((tmc1a_ra, tmc1a_dec))
+    if is_inside:
+        tmc1a_jwst.append(obs)
```

Comment:
This should be possible natively with astropy and regions; please use that rather than  brute force mpl.